### ISSUE TITLE
Not working right with updated comic

### ISSUE THREAD

**Issue by trlkly** [Questionable-Content-Extensions/client#25]
The new "now also on drip" link messes up the script, showing two copies of the name/location sidebar.  Seems the injection logic needs to be (slightly) changed. 

Blocking the aforementioned link (with uBlock) is viable workaround for now.

**Comment by ilyvion** [comment]
Hey, thanks for the heads up. I'll try to have a fix out sometime today.

**Comment by trlkly** [comment]
Confirming it works on my end, too, after I forced an update in TamperMonkey. 

It would be nice to be able to tell the user that there is an update available if they're using an old version, but I know that would take a lot of extra work.

**Comment by ilyvion** [comment]
Wouldn't actually be that much work, but I wish TamperMonkey's update cycle wasn't so slow, or that I could control it. I have a feeling it only really checks for updates when you start the browser, but if other people are like me, they only restart their computer (and thus browser) when it crashes or when they absolutely have to, which in some instances means weeks or months.

**Comment by ilyvion** [comment]
![image](https://user-images.githubusercontent.com/767490/46483789-c4a89700-c7f8-11e8-87dd-c9a82f311294.png)

Turns out the default is once per day. Which means a fix can take up to 24 hours to reach people.

**Comment by trlkly** [comment]
It says that, but I'm pretty sure it takes longer than that sometimes. Maybe those scripts have server issues or something.

Speaking of which, I wasn't sure if I should make a bug for this, but I notice your server seems to be a lot slower than it used to be--as in, it seems to take longer to pull in the tagged data. Before it was a second or two, at most, but now it can be upwards of 15 seconds (if the data isn't saved).

Anything you can do about that? It makes hunting through to find a particular comic in a location or with a certain character far more tedious.

**Comment by ilyvion** [comment]
> I notice your server seems to be a lot slower than it used to be--as in, it seems to take longer to pull in the tagged data. Before it was a second or two, at most, but now it can be upwards of 15 seconds (if the data isn't saved).

You are right. I looked into it, and the short of it is this:

If it's been a certain amount of time since a comic's data was last fetched, my service first checks QC itself to see if the news text has changed (since Jeph has a tendency to change his news texts a lot), and then returns the data as usual. (For the curious: Each such news update check has a time decay effect applied to it, meaning that over time, it'll take longer and longer between each time I check a comic's news text, until I finally just assume it's been set in stone. The more recent a comic is, the more often I check for updates.)

The problem comes in because my host recently changed something when it comes to DNS resolution and IPv6. In particular, it looks like it first resolves the IPv6 address and tries to connect using that (which times out at—you guessed it—15 seconds) before it moves on to using IPv4.

I probably should've been doing this news updating asynchronously anyway, but that didn't occur to me at the time, so this is where we're at. I've contacted my host to tell them about the problem, and hopefully they'll resolve it soon.
